### ISSUE TITLE
Goals for lensing sprint week

### ISSUE THREAD

**Issue by EiffL** [CosmoStat/jax-lensing#24]
This issue is to track our goals with @b-remy this week, towards getting as many results as we can for the paper:

Day 1: Tuesday
--------------------
@b-remy : will look at #12 
@EiffL : will look at #14 
-> Goal for the end of the day: Having tested sampling, and MAP in the Gaussian case, and being able to train a DeepMass model

Day 2: Wednesday
-------------------------
@b-remy : Will look at the MAP in the full non-Gaussian case (things to do for instance: PS as a function of iterations)
@EiffL : Will look at sampling in the full non-Gaussian case
-> Goal for the end of the day: Up to date script for sampling and MAP in full non Gaussian case.

Day 3: Thursday
----------------------
@b-remy : will look at #16 , #17 
@EiffL : will look at #15
-> Goal for the end of the day: Have the main numerical results/tests for the paper.

Day 4: Friday
------------------
@b-remy and @EiffL : Application to real COSMOS, and what to check....
@b-remy and @EiffL : Paper writing \o/

**Comment by NiallJeffrey** [comment]
I now have a python implementation of the Wiener posterior sampling (with a messenger field) for spin 2 fields with an assumed posterior. I can now convert in to a @jit jax function relatively easily. I'm busy most of today, but I have a few hours after 17h00 (France time).

**Comment by EiffL** [comment]
Oh awesome! Well, if you have time to add something like this in a branch, we'll be able to run some tests against the full Gaussian case, I hope we'll find that the mean of posterior samples matches your Wiener filter when we only assume the Gaussian prior. Also, am I wrong in thinking the MAP is equal to the posterior mean in the Gaussian case ?

**Comment by EiffL** [comment]
Ok, so for Day 1, I have a draft deepmass code, but haven't been able to test it yet :-| see here https://github.com/CosmoStat/jax-lensing/issues/14#issuecomment-784371819

**Comment by NiallJeffrey** [comment]
> Also, am I wrong in thinking the MAP is equal to the posterior mean in the Gaussian case ?

That's right - the posterior is symmetric Gaussian so the mean and MAP are the same

**Comment by NiallJeffrey** [comment]
I've started a new branch for the Wiener sampling https://github.com/CosmoStat/jax-lensing/tree/gaussian_prior. I've been delayed slightly by Jax versioning, but now up and running .

**Comment by EiffL** [comment]
Updated Plan for Thursday
------------------------------------
This is what we aim to do today

1. Merge #22 (@b-remy  and @EiffL )
2. @b-remy will put the trained network in a shared space
3. @b-remy adds a script for the MAP (or maybe add an option to retrieve the MAP in existing script) -> Should go in a new branch, looks like this goes towards #11 
4. @EiffL can look into annealed sampling in both Gaussian and Neural Score case, using the network trained by @b-remy 

-> Goal at the end of the day: 
- Both MAP and Sampling full functional in Gaussian case
- Try to have a working Neural Score sampling

**Comment by NiallJeffrey** [comment]
I have everything in place for Wiener sampling with the cosmos data. I have the initial code in:
     j`ax_lensing/wiener_sampling.py`
and I have shear data, true kappa and mask in 
     `data/ktng` and `data/COSMOS`
But, I am missing the noise model - specifically the noise covariance matrix

**Comment by b-remy** [comment]
This is the right place to find data, just mask sure you choose the file with 360 in their name to get maps 360x360.

For the noise model, we use a diagonal covariance matrix per component of shear with value `0.148`. We evaluated the noise on the COSMOS shear map here:
https://github.com/CosmoStat/jax-lensing/blob/training-script/notebooks/KappaTNG_COSMOS_emulation.ipynb

**Comment by EiffL** [comment]
Updated Plan for Friday
-------------------------------
This is what we aim to do today

Main targets:
1. @b-remy will look at adding an NFW object in a map #16 
2. @EiffL can look into annealed sampling in both Gaussian and Neural Score case, using the network trained by @b-remy

Bonus points:
1. MAP script
2. MAP with Neural Network

**Comment by NiallJeffrey** [comment]
@b-remy  in 
https://github.com/CosmoStat/jax-lensing/blob/training-script/notebooks/KappaTNG_COSMOS_emulation.ipynb the value seems not to be 0.148 -> `noise_dist_std = 0.2`

**Comment by b-remy** [comment]
@NiallJeffrey The data std is `0.148` as you measure. I did the computation in the section `Compare with True COSMOS shear noise` of the notebook.

`noise_dist_std = 0.2` is a parameter we use to train our denoiser, but not the exact std in the COSMOS map.
